lib/plugin-loader.ts

```diff
@@ -0,0 +1,853 @@
+    super(configuration, def);
+  }
+
+  public isValid(context: any) {
+    if (!(super.isValid(context) && this.filename 
+          && (this.authenticationCategory || this.authenticationCategories))) {
+      return false;
+    }
+    //we should not load authentication types that are 
+    //not requested by the administrator
+    if (!context.authManager.authPluginRequested(this.identifier,
+      this.authenticationCategory)) {
+        bootstrapLogger.warn("ZWED0029W"); //bootstrapLogger.warn("Authentication plugin was found which was not requested in "
+          //+ "the server configuration file's dataserviceAuthentication object. "
+          //+ "Skipping load of this plugin");
+      return false;
+    }
+    return true;
+  }
+
+  public exportDef() {
+    return Object.assign({}, super.exportDef(), {
+      filename: this.filename,
+      authenticationCategory: this.authenticationCategory
+    });
+  }
+  
+  public init(context: any) {
+    let filepath = path.join(this.location, 'lib', this.filename);
+    // Make the relative path clear. process.cwd() is zlux-app-server/bin/
+    if (!path.isAbsolute(filepath)) {
+      filepath = path.join(process.cwd(),filepath);
+    }
+    bootstrapLogger.info("ZWED0120I", this.identifier, filepath); //bootstrapLogger.log(bootstrapLogger.INFO,
+      //`Auth plugin ${this.identifier}: loading auth handler module ${filepath}`)
+    this.authenticationModule = require(filepath);
+    context.authManager.registerAuthenticator(this);
+  }
+}
+
+export class ProxyConnectorPlugIn extends Plugin {
+  private host: any;
+  private port: any;
+
+  constructor(configuration: any, def: any) {
+    super(configuration, def);
+    const remoteConfig = configuration.getContents(["remote.json"]);
+    if (remoteConfig) {
+      if (!this.host) {
+        this.host = remoteConfig.host;
+      }
+      if (!this.port) {
+        this.host = remoteConfig.port;
+      }
+    }
+  }
+  
+  public isValid(context: any): boolean {
+    if (!(super.isValid(context) && this.host && this.port)) {
+      return false;
+    }
+    //    //we should not load authentication types that are 
+    //    //not requested by the administrator
+    //    if (!context.authManager.authPluginRequested(this.identifier,
+    //      this.authenticationCategory)) {
+    //      bootstrapLogger.warn("Authentication plugin was found which was not requested in "
+    //          + "the server configuration file's dataserviceAuthentication object. "
+    //          + "Skipping load of this plugin");
+    //      return false;
+    //    }
+    return true;
+  }
+}
+
+const plugInConstructorsByType = {
+  "library": LibraryPlugIn,
+  "application": ApplicationPlugIn,
+  "windowManager": WindowManagerPlugIn,
+  "bootstrap": BootstrapPlugIn,
+  "desktop": DesktopPlugIn,
+  "nodeAuthentication": NodeAuthenticationPlugIn,
+  "proxyConnector": ProxyConnectorPlugIn
+};
+
+export class makePlugin {
```

Comment:
Not all functions are should be classes. This one appears to have been a factory function, which should have remained a function.